packages/plugin-ext/src/main/browser/quick-open-main.ts

```diff
@@ -13,346 +13,336 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
+/* eslint-disable @typescript-eslint/no-explicit-any */
 
 import { InputBoxOptions, QuickPickItem as QuickPickItemExt } from '@theia/plugin';
 import { interfaces } from '@theia/core/shared/inversify';
-import {
-    QuickOpenModel,
-    QuickOpenItem,
-    QuickOpenMode
-} from '@theia/core/lib/browser/quick-open/quick-open-model';
 import { RPCProtocol } from '../../common/rpc-protocol';
 import {
     QuickOpenExt,
     QuickOpenMain,
     MAIN_RPC_CONTEXT,
-    PickOptions,
-    PickOpenItem,
     TransferInputBox,
-    QuickInputTitleButtonHandle,
-    TransferQuickPick
+    TransferQuickPick,
+    TransferQuickPickItems,
+    TransferQuickInput
 } from '../../common/plugin-api-rpc';
-import { MonacoQuickOpenService } from '@theia/monaco/lib/browser/monaco-quick-open-service';
-import { QuickInputService, LabelProvider } from '@theia/core/lib/browser';
-import { PluginSharedStyle } from './plugin-shared-style';
-import { URI } from '@theia/core/shared/vscode-uri';
-import { ThemeIcon, QuickInputButton } from '../../plugin/types-impl';
-import { QuickPickService, QuickPickItem, QuickPickValue } from '@theia/core/lib/common/quick-pick-service';
-import { QuickTitleBar } from '@theia/core/lib/browser/quick-open/quick-title-bar';
+import { QuickInputService, QuickPickItem, QuickPickValue } from '@theia/core/lib/browser';
+import { QuickPickService } from '@theia/core/lib/browser/quick-input/quick-pick-service';
 import { DisposableCollection, Disposable } from '@theia/core/lib/common/disposable';
-import { QuickTitleButtonSide, QuickOpenGroupItem } from '@theia/core/lib/common/quick-open-model';
 import { CancellationToken } from '@theia/core/lib/common/cancellation';
+import { MonacoQuickInputService } from '@theia/monaco/lib/browser/monaco-quick-input-service';
```

Comment:
This was done similarly to 0.20:
![image](https://user-images.githubusercontent.com/71069170/117950842-44396380-b31c-11eb-97da-bbff47555c24.png)

I'm not fixed on this solution - but elaborate a better solution please.

Reply:
We need to use the monaco version of the service here because we need the proper return types, which are not define on the generic QuickInputService. As I've argued elsewhere, we should have a complete abstract set of services for QuickInput/Access and properly hide the monaco implementation, then we could just use QuickInputService here.

Reply:
closing this as this will be addressed in later PR for refactoring some parts of this PR.